workspaces/orchestrator/plugins/orchestrator-backend/src/service/api/v2.ts

```diff
@@ -141,9 +141,16 @@ export class V2 {
 
     let assessedByInstance: ProcessInstance | undefined;
 
-    if (includeAssessment && instance.businessKey) {
+    let variables: Record<string, unknown> | undefined;
+    if (typeof instance.variables === 'string') {
+      variables = JSON.parse(instance?.variables);
+    } else {
+      variables = instance.variables;
+    }
+
+    if (includeAssessment && variables?.orchestratorAssessmentInstanceId) {
       assessedByInstance = await this.orchestratorService.fetchInstance({
-        instanceId: instance.businessKey,
+        instanceId: variables.orchestratorAssessmentInstanceId as string,
```

Comment:
Do we need the `as string` here? I guess not thanks to the check on line 151

Reply:
line 151 helps for the case that it is undefined not for the type. because it is of type unknown (the field orchestratorAssessmentInstanceId) I get compile error. so, yes, it is needed.